lib/block-supports/layout.php

```diff
@@ -233,3 +233,36 @@ function( $matches ) {
 	remove_filter( 'render_block', 'wp_restore_group_inner_container', 10, 2 );
 }
 add_filter( 'render_block', 'gutenberg_restore_group_inner_container', 10, 2 );
+
+
+/**
+ * For themes without theme.json file, make sure
+ * to restore the outer div for the aligned image block
+ * to avoid breaking styles relying on that div.
+ *
+ * @param string $block_content Rendered block content.
+ * @param array  $block         Block object.
+ * @return string Filtered block content.
+ */
+function gutenberg_restore_image_outer_container( $block_content, $block ) {
+	$image_with_align = '/(^\s*<figure\b[^>]*)wp-block-image\s([^"]*((alignleft)|(alignright)|(aligncenter))(\s|")[^>]*>((.|\S|\s)*)<\/figure>)/U';
+
+	if (
+		'core/image' !== $block['blockName'] ||
+		WP_Theme_JSON_Resolver::theme_has_support() ||
+		0 === preg_match( $image_with_align, $block_content )
+	) {
+		return $block_content;
+	}
+
+	$updated_content = preg_replace_callback(
+		$image_with_align,
+		static function( $matches ) {
```

Comment:
What's the effect of the `static` keyword here? I thought static anonymous function declarations only mattered in the context of a class ([docs](https://www.php.net/manual/en/functions.anonymous.php#functions.anonymous-functions.static)).